### DISCUSSION TITLE
allow install via winget

### DISCUSSION THREAD

**Discussion by emhl** [garethgeorge/backrest#378] [Ideas]
WinGet is Microsofts official Package manager for Windows. 
it doesn't provide packaging for the applications. it is basically a collection of download links for .exe and .msi files

i have already tried to create a manifest using wingetcreate and it worked (the creation and the installation) without much hassle 
SmartScreen obviously complains that the application isn't signed, but apart from that the install is completely silent.

one thing that could maybe still cause some issues is, that the application reports as version 00.00.00.00

If you're fine with that i can submit the tested manifests to winget-pkgs under the Packet ID: garethgorge.Backrest

**Comment by garethgeorge** [comment]
Hi, thanks for reaching out -- I think this sounds very reasonable.

A winget package isn't on my roadmap at the moment, certainly happy for you to post a community maintained one. My understanding is that you PR the package to https://github.com/microsoft/winget-pkgs ? Feel free to CC me on that PR if you end up making one! And let me know if there's anything that needs doing on my end to fix the version reporting, is that being extracted from the installer? 

The only other potential issue I see is that backrest installs restic on first run which requires that it is run with administrator privileges. Not sure how often this is an issue but could prove problematic on some systems, you may want to test your manifest after first deleting the backrest managed version of restic so that it will be redownloaded.

**Reply by emhl** [reply]
yes. afaik that should be the process. 

the reported version is configured in the [install.nsi](https://github.com/garethgeorge/backrest/blob/0d98e4447b034c746c7c741fab0008e51d43730b/build/windows/install.nsi#L6C18-L6C29) file. Changing that variable should probably cause the correct version to be reported (i haven't tested this). There hopefully should be a way to load that variable from the environment instead so the file doesn't need to be changed on every release. 

I'll make sure to test the manifest after uninstalling backrest and cleaning up potentially not removed files. and up/downgrading an existing version instlled :D

**Reply by auanasgheps** [reply]
Hi @garethgeorge I'd love to get this package on Winget too, as I have submitted many packages to this repo in the past.
With the latest 0.17 the version is installed as 00.00.00.00. This needs to be fixed to get the package accepted, but I don't have the skills to do so.
 
The other is that the package isn't signed but it should not be a blocker.
